commit 56e098118a31753a9f755948bb5a47bc7111e214
Author: Andreas Schwab <schwab@suse.de>
Date:   Thu Aug 15 12:14:35 2019 +0200

    Update i386 libm-test-ulps

    Conflicts: ChangeLog removed

diff --git a/sysdeps/i386/fpu/libm-test-ulps b/sysdeps/i386/fpu/libm-test-ulps
index e83bae71b4..2232296fe0 100644
--- a/sysdeps/i386/fpu/libm-test-ulps
+++ b/sysdeps/i386/fpu/libm-test-ulps
@@ -1158,8 +1158,8 @@ float128: 4
 idouble: 4
 ifloat: 5
 ifloat128: 4
-ildouble: 7
-ldouble: 7
+ildouble: 8
+ldouble: 8

 Function: Imaginary part of "clog10_upward":
 double: 2
@@ -2222,8 +2222,8 @@ float128: 8
 idouble: 5
 ifloat: 5
 ifloat128: 8
-ildouble: 5
-ldouble: 5
+ildouble: 6
+ldouble: 6

 Function: "log":
 double: 1
